valohai_yaml/objs/pipelines/parameter.py

```diff
@@ -0,0 +1,68 @@
+from typing import TYPE_CHECKING, List, Optional
+
+from valohai_yaml.lint import LintResult
+from valohai_yaml.objs.base import Item
+from valohai_yaml.objs.utils import check_type_and_listify
+from valohai_yaml.types import LintContext, SerializedDict
+
+if TYPE_CHECKING:
+    from valohai_yaml.objs import Config, Pipeline
+
+
+def get_pipeline_parameter_target_node(target: str) -> str:
+    return target.split(".", 1)[0]
+
+
+def get_pipeline_parameter_target_parameter_name(target: str) -> str:
+    return target.split(".", 1)[1]
+
+
+class Parameter(Item):
+    """Represents a parameter definition within a pipeline definition."""
+
+    def __init__(self,
+                 *,
+                 name: str,
+                 targets: List[str],
+                 value: Optional[str] = None,
+                 default: Optional[str] = None,
+                 ) -> None:
```

Comment:
To be in harmony with the rest of the repo:
```suggestion
    def __init__(
        self,
        *,
        name: str,
        targets: List[str],
        value: Optional[str] = None,
        default: Optional[str] = None,
    ) -> None:
```